### PR TITLE
Add aggregate method to MongoDB

### PR DESCRIPTION
**Pull request by melohagan**
## Description
You can now use aggregations in the MongoDB connector.

There are two modes:
 - JSON
 - Pipeline

In the JSON mode, you simply provide an array of objects that represent each stage, as per the API docs: https://www.mongodb.com/docs/manual/aggregation/#aggregation-pipeline-example

In the Pipeline mode, you are presented with stages that can be added with the UI. This might be more suitable for "low-code users". 

Currently the json and pipeline modes store data separately, but it could be changed to allow mapping between the two.

Addresses: 
- https://github.com/Budibase/budibase/issues/5387

## Screenshots
**JSON mode**
Screenshot 2022-10-04 at 11 11 52

### Dropdown stages with templates:
mongo

### Swap stages
mongo

### CODE REVIEW

**Comment by Rory-Powell** [comment]
Nice work mel, this is definitely a more low-code solution. I was wondering about the new `Flow` configuration type, where is that set on the new UI? I wasn't sure from the screenshots, is it the `Action Type` select?

I was thinking the term Flow might be a bit of an overloaded, and possibly something that might make more sense is to display `Action Type` as `Query Type` and use `JSON` or `Form` as the values

**Comment by melohagan** [comment]
> Nice work mel, this is definitely a more low-code solution. I was wondering about the new `Flow` configuration type, where is that set on the new UI? I wasn't sure from the screenshots, is it the `Action Type` select?
> 
> I was thinking the term Flow might be a bit of an overloaded, and possibly something that might make more sense is to display `Action Type` as `Query Type` and use `JSON` or `Form` as the values

Changed 'flow' to 'pipeline' and renamed "Action Type" to "Query Type"

**Review by shogunpurple — APPROVED** [review]
LGTM! Great work Mel

File: packages/builder/src/components/integration/index.svelte [hunk 1]
```diff
@@ -24,6 +35,37 @@
...
+    }
+  }
+  function updateEditorsOnSwap(actionIndex, targetIndex) {
+    const target = query.fields.steps[targetIndex].value.value
+    stepEditors[targetIndex].update(query.fields.steps[actionIndex].value.value)
+    stepEditors[actionIndex].update(target)
+  }
+
+  function setEditorTemplate(fromKey, toKey, index) {
+    const currentValue = query.fields.steps[index].value.value
```
[the remark this answers is not in the document]
Comment (melohagan) [reply packages/builder/src/components/integration/index.svelte]: Added optional chaining to be sure!

**Pull request merged**